### ISSUE TITLE
tabs function getting passed nil

### ISSUE THREAD

**Issue by markleary** [chrekh/puppet-hosts#10]
I'm receiving the following error on a couple of my nodes (2 of 45):

```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Error while evaluating a Function Call, Failed to parse template hosts/hosts.erb:
  Filepath: /etc/puppetlabs/code/environments/production/modules/hosts/templates/hosts.erb
  Line: 6
  Detail: undefined method `length' for nil:NilClass
 at /etc/puppetlabs/code/environments/production/modules/hosts/manifests/init.pp:65:16 on node nodename
```

For some reason, the inline tabs function in hosts.erb is getting a nil/undef passed to it.  The strange thing is that I have nodes that receive identical configurations (aside from facts like fqdn) where all work but one.  And the relevant facts on the failing nodes look good.  One solution would be for tabs() to validate str, but I can't figure out where the nil value is coming from.

**Comment by chrekh** [comment]
Apparently one IP is nil, but I fail to imagine what could cause that.
Could you try to add a return unless str to tabs()
```ruby
def tabs(str)
  return "#bug " unless str
  count = ( 31 - str.length ) / 8
  return " " if count < 1
  return "\t" * count
end
```
This could help locate where the nil IP is comming from.

**Comment by markleary** [comment]
Ok, I see - turns out ipv6 is disabled on these hosts for some reason.  I missed this before, but the fact ipv6_lo_addrs is undef.  What I don't understand is ipv6_pri_addrs is undef on all my nodes, which does not cause a problem even though it appears you are using the same logic in the template.

The only params I am passing to the class are:

```puppet
  class { 'hosts':
    lo_names => ['localhost', 'localhost.localdomain'],
    entries     => { 'xxx' => [ 'xxx' ] }
  }
```

**Comment by chrekh** [comment]
The test for `elt.to_s == 'undef'` is supposed to reject all undefined IP:s but perhaps that is not enough?
I fail to recreate this, even if I deliberately add nil to addresses in list_addrs.rb

Could you try 23ad124db29f01917883f18eab0256893cdf76a8 and see if it helps?

**Comment by markleary** [comment]
That commit does resolve the issue for me, thanks!  For what it's worth, I'm using Puppet 4.9.4 on RHEL 6.9.

**Comment by chrekh** [comment]
Great.
I still don't understand why this is needed. But I don't think it can break anything either so let's just do it.
I have released a 2.3.1 with this included now.
